src/api/donations.py

```
@@ -0,0 +1,659 @@
+    try:
+        token = raw.removeprefix("Bearer ").strip()
+        dot = token.rfind(".")
+        if dot == -1:
+            return None
+        p, sig = token[:dot], token[dot + 1:]
+        exp = _hmac.new(secret.encode(), p.encode(), hashlib.sha256).hexdigest()
+        if not _hmac.compare_digest(sig, exp):
+            return None
+        padding = (4 - len(p) % 4) % 4
+        return json.loads(base64.b64decode(p + "=" * padding).decode())
+    except Exception:
```

Ruff (0.16.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)